Add format flag to listen cmd

gracegoo-stripe opened the pull request:
### Reviewers
r? @tomer-stripe 
cc @stripe/developer-products

 ### Summary
The --format flag is a string that currently only takes in JSON as a valid input option, similar to the log tailing command.
It's intended to replace the --print-json for the listen cmd. The difference in behavior between --format JSON and --print-json is that the new flag will stripe out all formatting in the payload. 

While working on the event streaming feature in VSCode, the stream was getting partial json because the json was being pretty-printed. This PR updates the listen cmd to behave the same way as the log tail command. 

### Testing
With the deprecated flag:
![Screen Shot 2021-03-17 at 9 54 54 AM](https://user-images.githubusercontent.com/75757829/111506458-e1a25d80-8706-11eb-84ae-853b10929832.png)

New flag:
![Screen Shot 2021-03-17 at 9 52 56 AM](https://user-images.githubusercontent.com/75757829/111506474-e49d4e00-8706-11eb-8443-437c0a7dbb88.png)

No flag:
![Screen Shot 2021-03-17 at 9 54 12 AM](https://user-images.githubusercontent.com/75757829/111506469-e36c2100-8706-11eb-843a-0b7a15fe1b2c.png)

vcheung-stripe commented on pkg/cmd/listen.go:
```diff
@@ -64,7 +65,11 @@ Stripe account.`,
 	lc.cmd.Flags().BoolVar(&lc.livemode, "live", false, "Receive live events (default: test)")
-	lc.cmd.Flags().BoolVarP(&lc.printJSON, "print-json", "j", false, "Print full JSON objects to stdout")
+	lc.cmd.Flags().BoolVarP(&lc.printJSON, "print-json", "j", false, "Print full JSON objects to stdout.")
+	lc.cmd.Flags().MarkDeprecated("print-json", " This flag is deprecated. Please use `--format JSON` instead")
```
Looks like the message `Flag --print-json is deprecated` is already printed.

vcheung-stripe commented on pkg/cmd/listen.go:
```diff
@@ -64,7 +65,11 @@ Stripe account.`,
-	lc.cmd.Flags().BoolVarP(&lc.printJSON, "print-json", "j", false, "Print full JSON objects to stdout")
+	lc.cmd.Flags().BoolVarP(&lc.printJSON, "print-json", "j", false, "Print full JSON objects to stdout.")
+	lc.cmd.Flags().MarkDeprecated("print-json", " This flag is deprecated. Please use `--format JSON` instead")
+	lc.cmd.Flags().StringVar(&lc.format, "format", "", `Specifies the output format of request logs
```
I think this might want to be
```diff
- Specifies the output format of request logs
+ Specifies the output format of events
```

vcheung-stripe reviewed (changes requested):
I have some suggestions for the output strings 😄

gracegoo-stripe commented:
> I have some suggestions for the output strings 😄

Oh my goodness, thank you for catching these!

vcheung-stripe reviewed (commented):
The messages look good to me. I can't speak for the rest of it so passing back r? @tomer-stripe

pepin-stripe commented:
LGTM but same questions as @tomer-stripe for case sensitivity. As a user I think I'd default to `--format json` since I'm usually used to pass in arguments in lower case

gracegoo-stripe commented:
> This looks good! Two quick questions:
> 
> 1. is `JSON` case sensitive? It probably shouldn't be but I can't remember what we do in request logs
> 2. did you test this with piping into `jq`? We should make sure that still works (and potentially mention that in the deprecation message too)

1. They are not but I can update them!
2. I did just now and that does work. I can suggest this in the deprecation message as well, that's a good idea. 
 
![Screen Shot 2021-03-18 at 9 39 05 AM](https://user-images.githubusercontent.com/75757829/111663358-f4cc3080-87cd-11eb-82b3-839b3f778401.png)

gracegoo-stripe commented:
Okay, I updated the flag check to be case insensitive for both `listen` and `logs tail`. Also added colorized output to the `listen` output because that looked fun:

![Screen Shot 2021-03-18 at 11 55 58 AM](https://user-images.githubusercontent.com/75757829/111682159-4500be00-87e1-11eb-8975-5851f39de811.png)

![Screen Shot 2021-03-18 at 11 53 28 AM](https://user-images.githubusercontent.com/75757829/111682171-46ca8180-87e1-11eb-8c75-8c42674323a4.png)

And the updated deprecation message:

![Screen Shot 2021-03-18 at 11 59 00 AM](https://user-images.githubusercontent.com/75757829/111682256-5d70d880-87e1-11eb-95c6-9a487db67afa.png)

pepin-stripe reviewed (approved):
LOVE IT!
